File: frontend/classes/PotentialAction.ts
interface PotentialActionData {
  id: number
  step: number
  faction: number
  type: string
  required: boolean
}

class PotentialAction {
  id: number
  step: number
  faction: number
  type: string
  required: boolean

  constructor(data: PotentialActionData) {
    this.id = data.id
    this.step = data.step
    this.faction = data.faction
    this.type = data.type
    this.required = data.required
  }
}

export default PotentialAction

File: backend/rorapp/serializers/potential_action.py
from rest_framework import serializers
from rorapp.models import PotentialAction


# Serializer used to read potential actions
class PotentialActionSerializer(serializers.ModelSerializer):
    
    class Meta:
        model = PotentialAction
        fields = ('id', 'step', 'faction', 'type', 'required')
